File: 10050 Hartals/solution.java
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
	    StreamTokenizer st = new StreamTokenizer(System.in);
	    st.nextToken();
	    int T = (int)st.nval;
	    StringBuilder output = new StringBuilder();
	    while(T-- > 0) {
	        st.nextToken();
    	    short N = (short)st.nval;
    	    st.nextToken();
    	    int P = (int)st.nval;
    	    short[] h = new short[P];
    	    for(int i = 0; i < P; ++i) {
    	        st.nextToken();
    	        h[i] = (short)st.nval;
    	    }
    	    int hartals = 0;
    	    do {
    	        if((N - 6) % 7 != 0 && N % 7 != 0) {
    	            for(int i = 0; i < P; ++i) {
    	                if(N % h[i] == 0) {
    	                    ++hartals;
    	                    break;
    	                }
    	            }
    	        }
    	    } while(N-- > 0);
    	    output.append(hartals + "\n");
    	}
        System.out.print(output);
	}
}

File: 10050 Hartals/solution.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

int main() {
	int T;
	scanf("%d", &T);
	while(T--) {
	    short N;
	    int P;
	    scanf("%hd %d", &N, &P);
	    short h[P];
	    for(int i = 0; i < P; ++i)
	        scanf("%hd", &h[i]);
	    int hartals = 0;
	    do {
	        if((N - 6) % 7 != 0 && N % 7 != 0) {
	            for(int i = 0; i < P; ++i) {
	                if(N % h[i] == 0) {
	                    ++hartals;
	                    break;
	                }
	            }
	        }
	    } while(N--);
	    printf("%d\n", hartals);
	}
	return 0;
}
